src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,96 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INNITIAL_CAPACITY = 16;
+    private static final float DEFAULT_FACTORY_LOAD = 0.75f;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INNITIAL_CAPACITY];
+        threshold = (int) (table.length * DEFAULT_FACTORY_LOAD);
+    }
+
+    public MyHashMap(int initialCapacity) {
+        if (initialCapacity < 0) {
+            throw new IllegalArgumentException("Initial capacity must be more than 0");
+        }
+        table = new Node[initialCapacity];
+        threshold = (int) (table.length * DEFAULT_FACTORY_LOAD);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        int index = getHash(key);
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value, null);
+        } else {
+            Node<K, V> current = table[index];
+            while (current != null) {
+                if (Objects.equals(current.key, key)) {
+                    current.value = value;
+                    return;
+                }
+                if (current.next == null) {
+                    current.next = new Node<>(key, value, null);
+                    break;
+                }
+                current = current.next;
+            }
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = getHash(key);
+        Node<K, V> currentNode = table[index];
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        private Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private <K> int getHash(K key) {
```

Comment:
my bad, didn't write to change the name of method 
as i can see, this method evaluate a position, not a hash, so:
```suggestion
    private <K> int getIndex(K key) {
```